server/webapp/WEB-INF/rails.new/app/presenters/api_v1/config/materials/package_material_representer.rb

```diff
@@ -15,21 +15,17 @@
 ##########################################################################
 
 module ApiV1
-    module Config
-        module Materials
-            class PackageMaterialRepresenter < ApiV1::BaseRepresenter
-                alias_method :material_config, :represented
-        
-                property :package_id, as: :ref, exec_context: :decorator
+  module Config
+    module Materials
+      class PackageMaterialRepresenter < ApiV1::BaseRepresenter
+        alias_method :material_config, :represented
 
-                def package_id
-                    material_config.getPackageId
-                  end
-
-                def package_id=(value)
-                    material_config.setPackageId(value)
-                  end
-              end
-          end
+        property :packageId, as: :ref, setter: lambda { |value, options|
+          package_definition = options[:config].getPackageRepositories().findPackageWith(value)
+          self.setPackageDefinition(package_definition) if (package_definition)
```

Comment:
Yes, throwing an exception will cause this to fail. When a null is returned, the validation takes care of adding appropriate error. I wouldn't go about throwing an exception in this case. And fixing the other method would have some implications to handle, for instance [this](https://github.com/gocd/gocd/blob/000fc0f7c60902dd05699a182310cfe4690b059c/server/webapp/WEB-INF/rails.new/app/controllers/admin/package_definitions_controller.rb#L79) does not handle null.